File: server/CSM Gate Foundation Server/CSM Gate Foundation Core/Services/Abstractions/Interfaces/IAuthService.cs
using CSM_Gate_Foundation_Core.Core.Models;
using CSM_Gate_Foundation_Core.Services.Models.Inputs;

using CSM_Server_Core.Abstractions.Interfaces;

namespace CSM_Gate_Foundation_Core.Services.Abstractions.Interfaces;

/// <summary>
///     Represents a server authentication service operations.
/// </summary>
public interface IAuthService
    : IService {

    /// <summary>
    ///     Authenticates a given credentials subscribing the session into the current <see cref="ISessionManager"/> context.
    /// </summary>
    /// <param name="input">
    ///     Authentication credentials.
    /// </param>
    /// <returns>
    ///     The <see cref="SessionData"/> information referencing the given <see cref="AuthInput"/> session.
    /// </returns>
    Task<SessionData> Authenticate(AuthInput input);
}

File: client/csm_gate_foundation_client/lib/src/services/abstractions/interfaces/iauth_service.dart
import 'package:csm_gate_foundation_client/csm_gate_foundation_client.dart';

/// Represents a server service communication handler for {Auth} operations.
abstract interface class IAuthService implements IService {
  /// Authenticates an user into [GateFoundationServer].
  ///
  /// [input] operation input parameters.
  Future<GateFoundationServerResolver<SessionData>> authenticate(AuthInput input);
}
